Sources/CvGame.cpp

```diff
@@ -3799,26 +3768,23 @@ int CvGame::countTotalNukeUnits()
 }
 
 
-int CvGame::countKnownTechNumTeams(TechTypes eTech)
+int CvGame::countKnownTechNumTeams(TechTypes eTech) const
 {
 	int iCount = 0;
 
-	for (int iI = 0; iI < MAX_TEAMS; iI++)
+	for (int iI = 0; iI < MAX_PC_TEAMS; iI++)
 	{
-		if (GET_TEAM((TeamTypes)iI).isEverAlive())
+		if (GET_TEAM((TeamTypes)iI).isEverAlive() && GET_TEAM((TeamTypes)iI).isHasTech(eTech))
 		{
-			if (GET_TEAM((TeamTypes)iI).isHasTech(eTech) && !GET_TEAM((TeamTypes)iI).isNPC())
```

Comment:
What is this change about? I can guess the ordering improves perf, but why `!isNPC` to `isEverAlive`?

Reply:
isEverAlive was the next if test inside that if,, he concatenated two ifs when he removed the isNPC check.
Since he change the for loop to stop before reaching the NPC teams there is no need to ask if the looped player is an NPC as the answer is already known.

Reply:
adding && and removing !isNPC() in the same place makes things look funky, but ya what toffer said. There was a short discussion on the forum a while back about some player and team loops having wrong size and stuff. That's a good example.

Reply:
Oh I see, I didn't know what MAX_PC_TEAMS was. Is PC meant to be Player controlled? Would be clearer if the name didn't have an ambiguous acronym in it.

Reply:
Playable Civilizaation/Character
As opposed to NPC (Non Playable Character)

Reply:
PC vs NPC is a pretty standard gaming naming convention, however it often usually considers NPC to be any non-human and we have actually considered the Non-Human players in the PC side as long as they are not a non-standard 'PLAYER' in the game.  NPCs are Barbs, Animals, etc...